### ISSUE TITLE
Proper language selection

### ISSUE THREAD

**Issue by AsafMah** [flide/8VIM#308]
**Is your feature request related to a problem? Please describe.**
More of something in the middle.
Today we have diacritics mode which probably works fine for, well, diacritics.

But I use the keyboard for Hebrew and English, and when it's a completely different alphabet, the current mode doesn't fit it very well:

1. Pressing shift once is practically useless, there aren't many cases you'd want only one letter in english
2. You can't really use shift properly in english mode

**Describe the solution you'd like**
An ability to select multiple layouts in the settings, and adding a "globe" button to the sidebar that will switch between them.
With that, also have a version of the hebrew layout without english letters (maybe numbers or symbols etc)

**Comment by ROBGUI09** [comment]
bump

**Comment by GRbit** [comment]
I find the 8vim keyboard fascinating and convenient, but since I type in two languages with a different alphabet it's a total pain to use it. Unfortunately, that downside makes the keyboard completely unusable for me, since I switch language very frequently. I can switch between different keyboards more easily than between languages in 8vim.

There are actually quite a lot buttons on the left? Do you think it could be possible to add layout switch button there? Or, considering that there is Ctrl button, Ctrl+Shift could switch the layout like it's done on PC.

Or am I missing something? Any other keyboard allow to switch language with one click, while with 8vim I could find ANY way to switch layout without going to settings. Is it supposed to be this way?

**Comment by yurikhan** [comment]
To whoever will be assessing importance and evaluating various possible solutions:

In a bilingual setting where the two languages use different scripts/alphabets, like English/Latin + Greek/Greek, or English/Latin + Ukrainian/Cyrillic, the need to switch languages can occur multiple times per sentence. Like, you’re writing a sentence in Hebrew, but you need to include the word “GitHub” in it, that’s two switches, one on each word boundary.

The vast majority of language switches will be on a word boundary; intra-word switches are somewhat rare but also possible. E.g. in inflectional languages (such as many Slavic ones), a proper name written in English/Latin can take on a native ending (joined by an apostrophe), like: *nom.* GitHub, *gen.* GitHub’а, *dat.* GitHub’у, *acc.* GitHub, *instr.* GitHub’ом, *prep.* о GitHub’е.

In summary: when you do need to switch languages, you need the switch gesture to be very easy. Fumbling in menus and settings dialogs does not cut it.

**Comment by BANanaD3V** [comment]
![image](https://github.com/user-attachments/assets/860abae1-1bd4-403e-bdde-181e5f0fc382)
what about this gesture to cycle between layouts? It's currently unused

**Comment by ManDay** [comment]
With the recent improvements to layout parsing, I think the question/problem of *how* we switch languages becomes moot; everyone will be able to pick whatever gesture they like best. The "only" thing that's missing currently seems to be the *Settings* UI to allow configuring multiple layouts/languages.

**Comment by dewdpol** [comment]
Maybe it is possible to replace radiobutton to checkbox in select language layout?

**Comment by ManDay** [comment]
That's a good idea, but it wouldn't get us all the way there, because that doesn't cover custom layouts (which for me, personally, is a must).

**Comment by dewdpol** [comment]
I don't use custom layout and i don't know "how it work" in this moment. Is it possible to make adding a custom layout to the "select language layout" list with the ability to replace / remove from the list "select language layout"? This may allow the use of a checkbox.

**Comment by katsu-chan** [comment]
> That's a good idea, but it wouldn't get us all the way there, because that doesn't cover custom layouts (which for me, personally, is a must).

I think many people use included layouts only, so the checkbox solution should be implemented first. And then @dewdpol 's solution should be implemented on top after.

**Comment by BeeFox-sys** [comment]
Agreed on custom layouts being important, i have a custom layout for english that works a bit better for me, and i also use the keyboard for toki pona of which there is no official layout. an option to load multiple custom layouts in order to select between them is a must for this feature.

**Comment by a-bezruchenko** [comment]
Bump. 

Much needed feature. Without this it doesn't even make sense to try this keyboard if you use two languages and often switch between them.
